### DISCUSSION TITLE
Is there any way to overwrite default style for certain objects?

### DISCUSSION THREAD

**Discussion by bolt12** [srid/emanote#438] [Q&A] [answered]
h2 headings appear inline and I'd like to fix that without me requiring to edit all classes for h2 headings manually when writing notes.

Example:
![image](https://github.com/srid/emanote/assets/20261421/d0896f41-c3ef-41bc-a80c-95dde9499fc6)
![image](https://github.com/srid/emanote/assets/20261421/d69f3a1b-6d5f-40c6-8304-3d4957fd3e09)

**Comment by srid** [comment]
Yes, you can copy https://github.com/srid/emanote/blob/master/emanote/default/templates/components/pandoc.tpl to your notebook's `templates/components/pandoc.tpl` and edit that.

But I wonder why the default h2 uses inline style in the first place. Can't we just remove it in the default? https://github.com/srid/emanote/blob/cedd7d8cde17e0dbe96b2b9d2210d8da7d2e2d01/emanote/default/templates/components/pandoc.tpl#L101

**Comment by bolt12** [comment]
Interesting, I couldn't find any info about that templates/components
folder. Is there any place I can look at?

Also yes I think  that inline style could be removed from the default :)

**Reply by srid** [reply, marked as answer]
Yea, it needs to be documented. At the moments only examples are available:

- https://github.com/srid/srid/tree/master/templates
- https://github.com/TheNeikos/hemera.systems/tree/master/content/templates
- https://gitlab.com/gil0mendes/website/-/tree/live/content/templates
- https://github.com/ChenghaoMou/chenghaomou.github.io/tree/master/templates

It basically works by doing a "union" of the default layer (in this github repo) and that of your notebook. So you can copy only the files you want to override. Kind of like unionfs; both default layer and your notebook are union mounted in Haskell using https://github.com/srid/unionmount

> Also yes I think  that inline style could be removed from the default :)

PR welcome! :)

**Reply by bolt12** [reply]
Thank you! I will look into it then 😄

**Comment by bolt12** [comment]
Another question: I noticed there's https://github.com/srid/emanote/blob/master/emanote/default/templates/filters/embed-note.tpl and https://github.com/srid/emanote/blob/master/emanote/default/templates/filters/embed-pdf.tpl. I know that if I do `![](path/to/pdf)` emanote renders the pdf in a neat way, but when I search for embed-pdf in the source code I don't see it being used. Is this some implicit behavior? Isn't this the template responsible for that neat pdf visualization?

I am asking this because I want to be able to do `![](/path/to/file.hs)` and have emanote render a nicely syntax highlighted code block. How would you suggest I do this?

**Reply by srid** [reply]
It is called here: https://github.com/srid/emanote/blob/24ff95d26ed8760248a0126a8727440f9059375a/emanote/src/Emanote/Pandoc/Renderer/Embed.hs#L66

> I am asking this because I want to be able to do ![](/path/to/file.hs) and have emanote render a nicely syntax highlighted code block. How would you suggest I do this?

Code embedding would be nice to have. See #24

Otherwise, I wonder if we should enable the user to embed *any* file type, possibly by extending this code: https://github.com/srid/emanote/blob/24ff95d26ed8760248a0126a8727440f9059375a/emanote/src/Emanote/Pandoc/Renderer/Embed.hs#L82-L95

**Reply by bolt12** [reply]
Thank you for such a prompt response, I'll put this on my TODOs I think it would be a nice addition. One could look at the embed template files and see if the file extension has any that matches.
